src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -649,6 +650,7 @@ type DiagnosticItem =
             | ErrorCode.AttributeInvalidOnCallable                -> "Invalid attribute placement. The attribute {0} cannot be attached to a callable declaration."
             | ErrorCode.UnresolvedTypeParameterForRecursiveCall   -> "The type argument(s) for the recursive call could not be inferred. Please provide explicit type arguments, e.g. Op<Int, Double>(arg)."
             | ErrorCode.TypeParameterResConflictWithTypeArgument  -> "The type of the expression needs to match the defined type argument. Expecting an expression of type {0}."
+            | ErrorCode.InvalidCyclicTypeParameterResolution      -> "Call cycle results in ambiguous or conflicting type parameter resolution."
```

Comment:
Is it possible to have the diagnostic message include the specific type variables that are ambiguous or conflicting?

Reply:
I had given some thought to how to do that, but wasn't able to come up with anything. I'd have to give this a lot more thought on what that would look like. I supposed it's not really a single type parameter causing the issue, just like it's not a single call that causes the issue. We would want to avoid having the cross product between all calls in the cycle and all type parameters in the cycle, but that might be doable. Unfortunately we are running short on time with this, so this is the simple solution for now, and I believe we will be able to update the error message later when we figure this out.

Reply:
OK, that seems reasonable, thanks for the explanation.